components/scream/src/share/atm_process/atmosphere_process.hpp

```diff
@@ -103,25 +116,36 @@ class AtmosphereProcess : public ekat::enable_shared_from_this<AtmosphereProcess
   // device and they are all 1d.
   // If the process *needs* to store the field as n-dimensional field, use the
   // template function 'get_reshaped_view' (see field.hpp for details).
-  // Note: this method will be called *after* set_grid, but *before* initialize.
+  // Note: this method will be called *after* set_grids, but *before* initialize.
   //       You are *guaranteed* that the view in the Field f is allocated by now.
   // Note: it would be tempting to add this process as provider/customer right here.
   //       However, if this process is of type Group, we don't really want to add it
-  //       as provider/customer. The group is just a 'design layer', and the stored processes
-  //       are the actuall providers/customers.
+  //       as provider/customer. The group is just a 'design layer', and the stored
+  //       processes are the actuall providers/customers.
   void set_required_field (const Field<const Real>& f) {
-    ekat::error::runtime_check(requires(f.get_header().get_identifier()),
-                         "Error! This atmosphere process does not require this field. "
-                         "Something is wrong up the call stack. Please, contact developers.\n");
+    ekat::error::runtime_check(
+        requires(f.get_header().get_identifier()),
+        "Error! This atmosphere process does not require this field. "
+        "Something is wrong up the call stack. Please, contact developers.\n");
     set_required_field_impl (f);
   }
   void set_computed_field (const Field<Real>& f) {
-    ekat::error::runtime_check(computes(f.get_header().get_identifier()),
-                         "Error! This atmosphere process does not compute this field. "
-                         "Something is wrong up the call stack. Please, contact developers.\n");
+    ekat::error::runtime_check(
+        computes(f.get_header().get_identifier()),
+        "Error! This atmosphere process does not compute this field. "
+        "Something is wrong up the call stack. Please, contact developers.\n");
     set_computed_field_impl (f);
   }
 
+  virtual void set_required_group (const ci_string_pair& /* group_and_grid */,
+                                   const std::set<Field<const Real>>& /* group */) {
+    ekat::error::runtime_abort("Error! This atmosphere process does not require a group of fields.\n");
```

Comment:
Maybe it would be good to explain that this is a constraint on process groups (if I've understood correctly)?

```
"Error! An atmosphere process group cannot require a group of fields.\n"
```

Reply:
Same thing for `set_updated_group` below.

Reply:
I just thought I'd add an empty default impl, so that all the atm proc (P3, SHOC, ...) do not need to have an empty impl of their methods. I can make the error message clearer, since it's a bit misleading. The issue is that if an atm proc exposes a non-trivial `get_required_groups` (i.e., returns a non-empty std::set), then it should also override this method to provide an impl of this method.

I'll update this.